external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -98,7 +100,7 @@ private[kafka010] class KafkaMicroBatchStream(
     } else if (minOffsetPerTrigger.isDefined) {
       ReadLimit.minRows(minOffsetPerTrigger.get, maxTriggerDelayMs)
     } else {
-      maxOffsetsPerTrigger.map(ReadLimit.maxRows).getOrElse(super.getDefaultReadLimit)
+      maxOffsetsPerTrigger.map(ReadLimit.maxRows).getOrElse(ReadLimit.allAvailable())
```

Comment:
why this change? Isn't it better to delegate to the parent if you have no opinion - e.g. the parent could use some config default or whatever.

Reply:
Previous comment seems to be out of date. It's to avoid Scala compiler bug.

While we are here, probably good to comment for the reason, and tag `// TODO: `

Reply:
https://github.com/apache/spark/pull/35238#discussion_r786502742

Reply:
I will add a comment

Reply:
how about fill a Spark Jira with the detailed description. Then we just need a single line comment here like:
```
TODO(SPARK-xxxx): directly call super.getDefaultReadLimit when ...
```

Reply:
sure